### PR TITLE
[11.0][FIX] API call to payments

### PR DESCRIPTION
**Pull request by robinkeunen**
The API nows only returns a single data dictionary instead of a list.
Also, the transaction id must be passed in the url.

Previously, `data` argument was overwritten in the function body. I'm tempted to either

- keep overwriting the `data` dict in the case some other function relies on the content of the data dict
- avoid overwriting the `data` dict as a best practice to avoid side effects.

I chose the latter option.

### CODE REVIEW

File: payment_mollie_official/models/mollie.py [hunk 1]
```diff
@@ -129,39 +134,33 @@ def _mollie_form_validate(self, data):
...
-        status = 'undefined'
-        for x in data_list:
-            if isinstance(x, dict) and x.get('id', False) and x['id'] == \
-                    transactionId:
-                data = x
-                break
-
-        if "status" in data:
-            status = data["status"]
+        status = mollie_response.get("status", "undefined")
```
Comment (remytms) [inline payment_mollie_official/models/mollie.py:163]: We may need to check that the result returned by mollie does match the transaction ID we sended. Like it's done on line 153 (at the end). Else we should raise an error and report it in the log.

File: payment_mollie_official/models/mollie.py [hunk 2]
```diff
@@ -129,39 +134,33 @@ def _mollie_form_validate(self, data):
...
 
         if status == "paid":
+            paidDatetime = mollie_response["paidDatetime"].replace(".0Z", "")
```
Comment (remytms) [inline payment_mollie_official/models/mollie.py]: standard naming for variable ? `paid_datetime` ? :)

File: payment_mollie_official/models/mollie.py (hunk 1 above)
Comment (remytms) [reply payment_mollie_official/models/mollie.py:163]: Resolved.

File: payment_mollie_official/models/mollie.py (hunk 2 above)
Comment (remytms) [reply payment_mollie_official/models/mollie.py]: Resolved

**Comment by robinkeunen** [comment]
@odoo-mvds can you have a look at this ? We've been using it in production for 3 months now.

**Comment by odoo-mvds** [comment]
Hello,

we do not support old versions anymore, however we will have a look at this fix as it's easy to merge.
We'll give you an update soon.

br,

**Comment by odoo-mvds** [comment]
Hi,

Thank you for your contribution and for submitting the pull request (PR).

As we have informed you, we are no longer maintaining the older version of the project to which your PR applies. We recommend upgrading to the latest version. In case we receive more such complains for the same issue, we will probably reopen the issue. Consequently, we will be closing this PR.

We truly appreciate your effort and interest in improving the project and look forward to any future contributions you may have.

Best regards,

**Pull request closed without merging**